modules/payloads/singles/linux/mipsbe/exec.rb

```diff
@@ -70,7 +72,9 @@ def generate(_opts = {})
     shellcode = shellcode + command_string + "\x00"
 
     # we need to align our shellcode to 4 bytes
-    (shellcode = shellcode + "\x00") while shellcode.length%4 != 0
```

Comment:
Bit of a weird behaviour change here between ruby versions:

```ruby
shellcode = ""

(shellcode = shellcode + "\x00") while shellcode.bytesize%4 != 0

puts RUBY_DESCRIPTION
puts shellcode.bytesize
```

Ruby 3.0:

```
$ ruby foo.rb       
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-darwin20]
0
```

Ruby 3.3:

```
$ ruby foo.rb
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-darwin22]
4
```

Reply:
Looks like this is fixed in the head of Ruby now; Will leave it as-is